spec/requests/requests_spec.rb

```diff
@@ -3,85 +3,243 @@
 require 'rails_helper'
 require 'telegram/bot/rspec/integration/rails'
 
-RSpec.describe 'Requests', telegram_bot: :rails do
+RSpec.describe 'Requests', type: :request do
   let(:organization) { create(:organization) }
 
-  describe 'POST /{organization_id}/requests' do
-    before(:each) { allow(Request).to receive(:broadcast!).and_call_original } # is stubbed for every other test
-    subject { -> { post organization_requests_path(organization, as: user), params: params } }
-    let(:params) { { request: { title: 'Example Question', text: 'How do you do?', hints: ['confidential'] } } }
-    let(:user) { create(:user, organizations: [organization]) }
+  describe 'GET /{organization_id}/requests' do
+    subject { -> { get organization_requests_path(organization, as: user), params: params } }
 
-    it { should change { Request.count }.from(0).to(1) }
+    let(:params) { {} }
 
-    it 'redirects to requests#show' do
-      response = subject.call
-      request = Request.last
-      expect(response).to redirect_to organization_request_path(organization, request)
-    end
+    context 'unauthenticated' do
+      let(:user) { nil }
 
-    it 'shows success notification' do
-      subject.call
-      expect(flash[:success]).not_to be_empty
+      it 'redirects to the sign in path' do
+        subject.call
+        expect(response).to redirect_to(sign_in_path)
+      end
     end
 
-    describe 'without hints param' do
-      let(:params) { { request: {  title: 'Example Question', text: 'How do you do?' } } }
-      it { should_not raise_error }
-    end
+    context 'unauthorized' do
+      let(:user) { create(:user, organizations: [create(:organization)]) }
 
-    describe 'without contributors' do
-      it { should_not raise_error }
+      it 'renders not found ' do
+        subject.call
+        expect(response).to be_not_found
+      end
     end
 
-    context 'with image file(s)' do
-      let(:params) do
-        { request: { title: 'Message with files', text: 'Did you get this image?', files: [fixture_file_upload('profile_picture.jpg')] } }
+    context 'authenticated and authorized' do
+      let(:user) { create(:user, organizations: [organization]) }
+
+      before do
+        create(:request, title: 'I belong to the organization', organization: organization)
+        create(:request, title: 'I do too!', organization: organization)
```

Comment:
I would propose to hold the specs as concise and short as possible. In this regard: Can we remove any of these stubs without loosing sth. logic wise?

Reply:
No.